python/pyspark/pandas/tests/data_type_ops/test_boolean_ops.py

```diff
@@ -24,289 +24,273 @@
-                if isinstance(psser.spark.data_type, BooleanType):
-                    self.assert_eq(self.pser + pser, (self.psser + psser).sort_index())
-                else:
-                    self.assertRaises(TypeError, lambda: self.psser + psser)
+        pdf, psdf = self.pdf, self.psdf
+
+        b_pser, b_psser = pdf["bool"], psdf["bool"]
+        self.assert_eq(b_pser + 1, b_psser + 1)
+        self.assert_eq(b_pser + 0.1, b_psser + 0.1)
+        self.assert_eq(b_pser + b_pser.astype(int), b_psser + b_psser.astype(int))
+        self.assert_eq(b_pser + b_pser, b_psser + b_psser)
+        self.assert_eq(b_pser + True, b_psser + True)
+        self.assert_eq(b_pser + False, b_psser + False)
+
+        for col in self.numeric_df_cols:
+            pser, psser = pdf[col], psdf[col]
+            self.assert_eq(b_pser + pser, b_psser + psser)
+        for col in self.non_numeric_df_cols:
+            pser, psser = pdf[col], psdf[col]
+            if col == "bool":
+                self.assert_eq(b_pser + pser, b_psser + psser)
+            else:
+                self.assertRaises(TypeError, lambda: b_psser + psser)
 
     def test_sub(self):
-        pser = self.pser
-        psser = self.psser
-        self.assert_eq(pser - 1, psser - 1)
-        self.assert_eq(pser - 0.1, psser - 0.1)
-        self.assert_eq(pser - pser.astype(int), psser - psser.astype(int))
-        self.assertRaises(TypeError, lambda: psser - psser)
-        self.assertRaises(TypeError, lambda: psser - True)
+        pdf, psdf = self.pdf, self.psdf
+
+        b_pser, b_psser = pdf["bool"], psdf["bool"]
+        self.assert_eq(b_pser - 1, b_psser - 1)
+        self.assert_eq(b_pser - 0.1, b_psser - 0.1)
+        self.assert_eq(b_pser - b_pser.astype(int), b_psser - b_psser.astype(int))
+        self.assertRaises(TypeError, lambda: b_psser - b_psser)
+        self.assertRaises(TypeError, lambda: b_psser - True)
 
-        with option_context("compute.ops_on_diff_frames", True):
-            for pser, psser in self.numeric_pser_psser_pairs:
-                self.assert_eq(self.pser - pser, (self.psser - psser).sort_index())
+        for col in self.numeric_df_cols:
+            self.assert_eq(b_pser - pdf[col], b_psser - psdf[col])
 
-            for psser in self.non_numeric_pssers.values():
-                self.assertRaises(TypeError, lambda: self.psser - psser)
+        for col in self.non_numeric_df_cols:
+            self.assertRaises(TypeError, lambda: b_psser - psdf[col])
 
     def test_mul(self):
-        pser = self.pser
-        psser = self.psser
-        self.assert_eq(pser * 1, psser * 1)
-        self.assert_eq(pser * 0.1, psser * 0.1)
-        self.assert_eq(pser * pser.astype(int), psser * psser.astype(int))
-        self.assert_eq(pser * pser, psser * psser)
-        self.assert_eq(pser * True, psser * True)
-        self.assert_eq(pser * False, psser * False)
-
-        with option_context("compute.ops_on_diff_frames", True):
```

Comment:
The change looks good but can we keep (or separate) a basic set of test with `compute.ops_on_diff_frames` option on? Feel free to do it in a separate PR if you would like to.

Reply:
https://github.com/apache/spark/blob/master/python/pyspark/pandas/tests/test_ops_on_diff_frames.py are with `compute.ops_on_diff_frames` option on. Would you think that's enough? If not, I would like to follow up on this.